ydb/core/kqp/provider/rewrite_io_utils.cpp

```diff
@@ -15,31 +15,54 @@ using namespace NNodes;
 
 constexpr const char* QueryGraphNodeSignature = "SavedQueryGraph";
 
-NSQLTranslation::TTranslationSettings CreateViewTranslationSettings(const TString& cluster) {
+NSQLTranslation::TTranslationSettings CreateViewTranslationSettings(const TString& cluster, EKikimrQueryType queryType) {
     NSQLTranslation::TTranslationSettings settings;
 
     settings.DefaultCluster = cluster;
     settings.ClusterMapping[cluster] = TString(NYql::KikimrProviderName);
     settings.Mode = NSQLTranslation::ESqlMode::LIMITED_VIEW;
 
+    settings.Flags.insert("FlexibleTypes");
```

Comment:
These flags are copied from the default [settings](https://github.com/ydb-platform/ydb/blob/2d8ae71907f4da2c07c941ea8efa277d4680d216/ydb/core/kqp/host/kqp_translate.cpp#L107) that KQP host is using to compile queries.

The following settings are deliberately left out (i.e. not copied from KQP host):
- PG (postgres) settings (PG syntax is not yet officially supported in combination with views)
- SQL version settings (I imagine, views don't have to support v0 syntax)
- external data source settings (they haven't been supported inside views yet)
- EndOfQueryCommit setting, because I believe views should not manage commits by themselves
- QueryParameters (views don't support parameters)
- ApplicationName (to be frank, I have no idea, what use does it have)
- GUCSettings (they are used in PG syntax mode that is not yet officially supported in combination with views)

One important thing to note here is that the flags (i.e. pragmas) depend on the query type used to compile the `SELECT` from a view. This query type might differ from the query type that was used to `CREATE VIEW`. This might lead to the results of internal view queries being dependent on the particular query type that is being used at the time of a `SELECT` from the view. This might be considered a bug, because a view should produce the same result whenever it is selected from. It should act almost like a real table that has its rows saved and persisted on disk to never change unless these rows change in the underlying tables from which the view selects from.

This "bug" will be fixed later, when we will decide which parts of the `CREATE VIEW` context to save inside the view.

Reply:
1. We can reuse the basic settings of KQP and remove unsuitable fields. Do we really need to keep two independent versions of TTranslationSettings?
2. What do we need to do to support external data sources? What is the difference between data shards and external data sources in case of views?

Reply:
1. We could reuse `TKqpTranslationSettingsBuilder` to compile the view query if that is what you suggest. However, the ideal way to get `TTranslationSettings` for the view query compilation is to fetch them from the parsing context of the `CREATE VIEW` statement and persist them in the body of the view. We should not use effective `TTranslationSettings` at the moment of a `SELECT` from a view, because they might differ from those that were in effect during `CREATE VIEW` statement. (For example, `query type` might differ leading to default pragmas being different.)

2. We haven't tested this scenario at all. I'm sure there will be some hidden differences that will become a problem for views in combination with external data sources. We will probably think about this scenario later, after the first release of views.

Reply:
1. My proposal is to try to reuse the part of settings between TKqpTranslationSettingsBuilder and CreateViewTranslationSettings. Also, sometimes we want to override the persisted view translations settings. For example, we want to display a warning if the user uses some expression inside the view. We have already used this approach for bindings. `EBindingsMode BindingsMode;` prefix. 
2. We think about using the view inside YQ service and there we use the external sources. And it was interesting to know the status in this part. I got the answer here, thank you)

Reply:
1. In some sense, I have done exactly that: reused (copy-pasted) some parts of the settings that TKqpTranslationSettingsBuilder builds. I plan to persist translation settings inside a view at the moment of the view creation for later use during `SELECT` from the view. It would eliminate the need to build the settings here, during `RewriteReadFromView`. We would just take the persisted ones and use them.

    As for overriding of the settings to display a warning if some forbidden expression is used inside a view, we can do this at the moment of persisting of the settings during `CREATE VIEW`. Or are you trying to say that we might someday need to override the settings of some old view that was created years ago to forbid some discontinued expression inside it? Well, in that case it seems suitable to override it inside the dedicated, view-specific `CreateViewTranslationSettings` function and don't mix it with the general `TKqpTranslationSettingsBuilder`.

    And lastly, all I am trying to say is that I will rewrite `CreateViewTranslationSettings` function later. The changes in this function that are introduced in this PR are temporary. It is better to think all of this through when we start implementing `TTranslationSettings` persisting during `CREATE VIEW`.

Reply:
Used TKqpTranslationSettingsBuilder in the latest [revision](https://github.com/ydb-platform/ydb/blob/fee0d0ddce1cbf573356840741486942e08e697c/ydb/core/kqp/provider/rewrite_io_utils.cpp#L24)